### PR TITLE
Use ts strict checking for binding mqtt

### PR DESCRIPTION
**Pull request by egekorkan**
I had started on this but did not manage to move forward due to time constraints. I still wanted to make it public so that others can take the work.

@hasanheroglu maybe you can have a quick look? Otherwise, I will try to get back to it in two weeks :)

### CODE REVIEW

File: packages/binding-mqtt/src/mqtt-client.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ export default class MqttClient implements ProtocolClient {
         this.scheme = "mqtt" + (secure ? "s" : "");
     }
 
-    private client: mqtt.MqttClient = undefined;
+    private client: mqtt.MqttClient | undefined = undefined;
```
Comment (JKRhb) [inline packages/binding-mqtt/src/mqtt-client.ts:43]: I think this could be simplified a bit :)

```suggestion
    private client?: mqtt.MqttClient;
```

**Comment by relu91** [comment]
@danielpeintner should be ready to go. Before merging I'll do some rebase to clean up the commit messages.

**Review by danielpeintner — APPROVED** [review]

**Comment by relu91** [comment]
argh I forgot to squash :/

**Comment by egekorkan** [comment]
Thanks for doing it! :)

**Pull request merged**